src/operator/matrix_op-inl.h

```diff
@@ -643,6 +654,10 @@ void Slice(const TBlob &src,
   mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
   int ndim = static_cast<int>(ret->shape_.ndim());
 
+  int axis_size = static_cast<int>(src.shape_[param.axis]);
+  param.begin = (param.begin < 0) ? (param.begin + axis_size + 1) : param.begin;
```

Comment:
One way I can think of is to use end=None for end=n
instead of using param.Init you can mannually parse kwargs to see if 'end' is present
or you can set default value for end to INT_MAX